Persist IOU data in Onyx

Jag96 opened the pull request:
### Details
This PR adds a new API command for `GetIOUReport` and starts persisting the IOUReport collection in Onyx.

### Fixed Issues
Fixes https://github.com/Expensify/Expensify/issues/157104

### Tests
1. On web, log out of e.cash and open the console (cmd + option + j)
2. Open the `Application` tab and select `localhost:8080` under `Local Storage`
![image](https://user-images.githubusercontent.com/3981102/111380061-1cea5100-8661-11eb-8aed-40d403d1b945.png)
3. Log into an account without an IOU and check the LocalStorage keys, confirm there is nothing shown if you filter by `reportIOU`
![image](https://user-images.githubusercontent.com/3981102/111380304-65097380-8661-11eb-8f1d-9b95f562cf9b.png)
4. Use the `CreateIOUTransaction` Curl request from https://github.com/Expensify/Expensify/issues/157104#issue-829068854 to create an IOU transaction. Ensure the debtorEmail is different from the email/accountID/authToken you set in the cookie field
5. Log into cash as the account of the person who created the IOU. Confirm that you see a value in localStorage for `reportIOU_` that contains the IOU reportID. For an example of the full data, see https://github.com/Expensify/Expensify/issues/157104#issuecomment-799886417
<img src='https://user-images.githubusercontent.com/3981102/111383988-3641cc00-8666-11eb-976f-daa3ac923cb3.png' width='500'/>

6. Log out, and log into cash using the debtorEmail value you used in step 4, confirm you see a value in localStorage for `reportIOU_` that contains the IOU reportID
7. Create an IOUTransaction from an account A that has been the only one to comment on a chat report and User B is the debtor.
8. Confirm that when logging in with UserA, the reportIOU collection contains that report.

Additionally, confirm that there are no regressions with sending messages to other accounts

### Tested On

- [x] Web
- [ ] Mobile Web
- [x] Desktop
- [x] iOS
- [x] Android

### Screenshots
N/A

Jag96 committed:
add GetIOUReport api command

Jag96 committed:
start persisting IOUReport data in onyx

Jag96 commented on src/libs/actions/Report.js:
```diff
@@ -495,6 +495,85 @@ function fetchChatReports() {
+        .value();
+
+    // If we have more than one participant, this is not an IOU
+    if (otherParticipants.length > 1) {
```
cc @Julesssss just double-checking that it isn't possible for an IOU to have more than 2 participants. The API command for `getIOUReport` only takes one debtorEmail atm

Julesssss replied:
That's right, each IOU report has a 1:1 relationship. Bill splits are a collection of 1:1 IOU Reports.

Julesssss reviewed (dismissed):
Looks great, approving for now in case no additional changes are pushed 👍

tgolen commented on src/libs/API.js:
```diff
@@ -621,6 +620,17 @@ function Mobile_GetConstants(parameters) {
+ * @returns {Promise}
+ */
+function GetIOUReport(parameters) {
+    const commandName = 'GetIOUReport';
```
Hm, I would have really liked for this API to be called `GetIOUReportID` since it's only returning the report ID and nothing else. It would be even better if this API would remain named `GetIOUReport` and return the entire report object that is later requested in a second API request (get stuff).

iwiznia replied:
hmmm maybe yeah, let's keep as is and we can do that as an optimization later.

tgolen replied:
No, I don't think this is too much of a pre-optimization to ask for. This isn't about optimizing, this is about doing it the correct way in the first place. We should never settle for making two API calls when only one API call will do. Since this is building a new feature, let's do it correctly rather than to force ourselves to have future work.

iwiznia replied:
ok, sending a PR that's backward compatible with this. We can wait for it and use it in this PR or probably better, merge this as is to unblock other issues and use the new data in a followup PR.

tgolen replied:
I'm OK implementing it in another PR.

iwiznia replied:
PRs here:
https://github.com/Expensify/Web-Expensify/pull/30479 
https://github.com/Expensify/Auth/pull/5368

Jag96 replied:
Once https://github.com/Expensify/Web-Expensify/pull/30479 hits production I can create a follow-up for this

iwiznia replied:
This is not live, so we can do that now I think...

Jag96 replied:
After some testing, it looks like the changes in https://github.com/Expensify/Web-Expensify/pull/30479 aren't returning the transactionList, so we shouldn't update this to remove the GetReportStuff call yet. Once I've updated GetIOUReport to also return the transactionList (looking into it), then I can create a follow-up here.

Jag96 replied:
Issue created here: https://github.com/Expensify/Expensify/issues/157722

Julesssss replied:
Looks like this thread can be resolved

tgolen commented on src/libs/actions/Report.js:
```diff
@@ -495,6 +495,85 @@ function fetchChatReports() {
+        iouReportID = response.reportID;
+
+        Log.info('[Report] Fetching IOU report data', true, iouReportID);
+        return API.Get({
```
Yeah, it's kind of a shame we have to make dual requests. Since this is a new feature, I say we take the time to do it properly and just have this data returned from `GetIOUReport`.

Jag96 replied:
@Julesssss do you have any reason why we shouldn't update `GetIOUReport` to just return the data from `GetReportStuff` on the API side? This sounds like a good improvement to me

Julesssss replied:
Nope, that seems like a good improvement to me too, I recall DD discussions about this and other improvements... I believe the counter argument was that it was an early optimisation that could come later.

CC'ing @iwiznia to get your thoughts here

iwiznia replied:
Let's not block on this, yes, we are doing 2 requests but that's not that bad. We can optimize this later.

tgolen commented on src/libs/actions/Report.js:
```diff
@@ -495,6 +495,85 @@ function fetchChatReports() {
+    }).then((response) => {
+        iouReportID = response.reportID;
+
+        Log.info('[Report] Fetching IOU report data', true, iouReportID);
```
I don't think we need this log here. It's apparent from the network tab of the JS console what's being done.

tgolen commented on src/libs/actions/Report.js:
```diff
@@ -495,6 +495,85 @@ function fetchChatReports() {
+        });
+    }).then((response) => {
+        const iouReportData = response.reports[iouReportID];
+        Onyx.merge(`${ONYXKEYS.COLLECTION.REPORT_IOUS}${iouReportID}`,
```
Is there a good reason to put these into their own Onyx collection? I kind of think that they should just be treated like normal reports. If anything, we should maybe add a `type` property to the report collection to be able to tell the different report types apart in the future (I think this will scale better as we get more report types). Note: this will also require writing a migration so that chat reports are stored with the proper `type` and we _maybe_ need to filter the LHN so that it only displays reports with the chat type?

Jag96 replied:
So wherever we'd use the REPORT_IOUS collection we could just connect to the REPORT collection instead and filter by type in the callback? This makes sense to me, @Julesssss is there a reason why a new collection would work better here?

iwiznia replied:
What's the benefit? I thought this was better because it allowed us to subscribe to just IOUs... are you suggesting we save them with a key like `REPORTS_${type}_${reportID}`?

Julesssss replied:
The idea was that keeping a layer of separation between them led to a simpler implementation, as both the method of retrieving and the persisted fields are quite different for Chat/Reports. That was the only reason though.

I'd like to think this through a bit first, to make sure we don't create new problems with this approach.

tgolen replied:
> What's the benefit? 

This same question can be asked of the current implementation. Currently, I don't see that we are subscribing to a list of all IOU reports, we are only accessing a single one. Are there plans to subscribe to the whole list of _just_ IOU reports? If not, then I don't think we need to optimize for that case.

> are you suggesting we save them with a key like REPORTS_${type}_${reportID}?

No, I wasn't suggesting that because this is the current implementation. I'm suggesting that every report (regardless of type) be stored as `REPORT_${reportID}` and then each report has a property for `type`. 

> One concern is that we end up with a mega collection of every report, chat, invoice, IOU... and that we'll make each of these individual cases slower.

I don't see this as a real concern or it's a pre-optimization. Those cases don't exist today and we shouldn't speculate about what would slow them down or not. It's not hard to slap a `_.findWhere(reports, {type:'chat'})` in the places that you need specific report types.

My biggest concern is with this getting out of hand at some point. I am worried that will can lead to a separate action file for each report type that still shares 95% of things in common with each other and a whole bunch of different Onyx collections whose objects are 95% the same. I think that will make the entire system more difficult to learn and maintain over time.

iwiznia replied:
Yeah, I don't know what will be best, since we don't know what future uses we will have.
We should do what's easier to change later (which honestly I have no idea what that is).

tgolen replied:
I think if I had to pick an "easier" path, it's probably the one we are currently on because it would mean that no Onyx migration needs to be written in this PR and doesn't require any additional code changes for something that none of us are totally clear on.

Jag96 replied:
Since we're currently only fetching [chatReports](https://github.com/Expensify/Expensify.cash/blob/ab38d478ac912b06529bd0e1b1eb5381cbdce8e2/src/libs/actions/Report.js#L479), if we decide to keep everything in reports we'll have to update that logic on the backend to now fetch IOUs and Chat reports, and add necessary type filtering throughout the codebase. Then later on, if we want to migrate to moving everything to a `reportIOU` collection, we'd have to write a migration that removes any IOU reports from Onyx, re-fetches them via GetReportStuff (or some other method), and then inserts the simple structure into Onyx for that collection key. Additionally, we'd have to go through and clean up any filtering logic.

Alternatively, if we keep the `reportIOU` collection and later decide to move everything back into the report collection, we'd need to make the back-end updates and add filtering, as well as add a migration that would remove the reportIOU collection from Onyx, re-fetch all of those reports and add the simplified data to the report collection.

Both of these require some work and a migration, but both options are reasonable/doable IMO. I understand @tgolen's concern about having different collections for different report types that could easily hold the same data, so my main question is if there are any other reasons to use a new collection other than the fear that using the report collection will slow things down (other than the fact that this code is written already).

Julesssss replied:
>  so my main question is if there are any other reasons to use a new collection

The only other reason I see is that few fields will be shared for now. There is no need to persist the full IOU report and perhaps option B would require us to do this to align with reports? 

I lean towards the current solution mainly to speed up the IOU project progress, but don't feel too strongly either way. Let's decide on one path and keep moving forward. Please choose from **A** or **B**:

**A:** Keep separate collection for now:
- Quicker and arguably simpler *for now
- A pre-optimization
- Persisted IOUReport data remains lean
- No Onyx migration *for now

**B:** Merge IOU data with the report collection:
- Leads to more shared functionality between Report types
- Better solution in the long run
- Will need to update backend logic to now fetch IOUs and Chat reports
- Might need to persist full IOU Report data (unconfirmed, not a huge problem)
- Requires Onyx migration

Jag96 replied:
After bringing this conversation to [slack](https://expensify.slack.com/archives/C011W8BJ9L6/p1616003536027800), the general consensus was that we'll go with option A here.

tgolen commented on src/libs/actions/Report.js:
```diff
@@ -529,6 +608,8 @@ function fetchActions(reportID, offset) {
                 .max()
                 .value();
 
+            updateIOUReportData(reportID, indexedData);
```
I don't really understand the first parameter here. Is it used? Why would it be passing a `reportID` only to call the API to get a different `reportID` to fetch with `API.GetIOUReport()`?

Jag96 replied:
Ah yes, this needs to be removed

tgolen commented on src/libs/actions/Report.js:
```diff
@@ -495,6 +495,85 @@ function fetchChatReports() {
+    }
+
+    // If we have an IOU action, get the IOU reportID
+    let iouReportID = 0;
```
Yeah, I'm really confused here. If `reportID` is being passed to this method, why are we looking up a different report ID and why does that report have different data than the original report?

Julesssss replied:
The `reportID` param is the chat report ID, IOUTransactions are stored as expenses on a separate IOU Report, allowing us to close and switch ownership without tying it too closely to the ChatReport. This has always been the plan.

tgolen replied:
OK, thanks for explaining that. I think it's good context to have, but definitely not apparent from just looking at the code so it will probably confuse others as well. Typically, the best way to combat that is with better code comments that give a bit more context. Would you mind adding a few more comments here to help?

Jag96 replied:
I've updated the comment there, let me know if that still doesn't make sense!

Jag96 committed:
remove unused param, add comments

Jag96 commented:
Updated, https://github.com/Expensify/Expensify.cash/pull/1785#discussion_r595296484 is the only unresolved thread at the moment

Jag96 commented:
https://github.com/Expensify/Expensify.cash/pull/1785#discussion_r595296484 has been resolved, this is ready for review again!

Jag96 committed:
remove unnecessary param

Julesssss commented:
Looking good, but I noticed two issues when running the test:

1) I'm seeing an extra undefined 'actorEmail' in `otherParticipants`
![Screenshot 2021-03-18 at 13 07 44](https://user-images.githubusercontent.com/10736861/111631128-f5bf8b00-87ea-11eb-98d0-01eeb45bf96c.png)
This makes the `(otherParticipants.length > 1)` condition true, preventing the `getIOUReport` request. Once I added `.without(undefined)`, I was able to verify that the data was persisted to Onyx. _Didn't look further into why this occurs or a better fix._

2) I'm only seeing IOU data when logged into the account of the IOU creator
I think this is because we're using `actorEmail`, which is the same value for both users. Perhaps we should use `managerEmail` or `ownerEmail` (whichever does not match the current user). These will swap depending on who owes who at the current point in time, but both emails should be available this way.

Julesssss reviewed (changes requested):
Looking good, but a couple of issues mentioned above need addressing

tgolen commented on src/libs/actions/Report.js:
```diff
@@ -495,6 +495,84 @@ function fetchChatReports() {
+        .pluck('actorEmail')
+        .unique()
+        .without(currentUserEmail)
+        .value();
```
@Julesssss my guess is this is where the `undefined` participant is coming from. I think there needs to be a `.flatten()` added into the end of this chain to remove it (this is pretty typical).

Jag96 commented:
> I'm seeing an extra undefined 'actorEmail' in otherParticipants
I'm unable to reproduce this, what is the account setup you have for IOU testing?

> I'm only seeing IOU data when logged into the account of the IOU creator
I'm also unable to reproduce this, I see the IOU data get populated when logged in as both the IOU creator and the debtor. I will try testing with some additional cases (no report_actions from the debtor, create a report with IOUs from both parties) to try to reproduce.

That testing brings up a good point however, since we're using the actorEmail to build the list of participants we may not be able to get the debtorEmail if the only person who has chatted on the report is the requestor. I'll have a look at that test case and if it is a problem, look into getting the debtorEmail another way.

Jag96 commented:
After some testing, I confirmed the issues above exist if the requestor is the only one with chats on a chat report. Therefore, using the actorEmail values in the reportHistory isn't enough to get the participants on a report.

> Perhaps we should use managerEmail or ownerEmail (whichever does not match the current user)

The problem with this is that we don't have a managerEmail or ownerEmail in the response from getHistory. Additionally, [fetching actions](https://github.com/Expensify/Expensify.cash/blob/d27bed5759c35c6ddf89cfc8c6cbe80d17ea02c8/src/libs/actions/Report.js#L640) happens separately from where we fetch the [rest of the report data](https://github.com/Expensify/Expensify.cash/blob/d27bed5759c35c6ddf89cfc8c6cbe80d17ea02c8/src/libs/actions/Report.js#L489), so we don't have access to the participants inside getActions where we're filtering by the "IOU" actionName.

I looked into creating a local object that keeps track of the participants of each report, similar to how we keep track of lastReadSequenceNumbers or reportMaxSequenceNumbers. Then, in handleReportChanged, I tried setting the value of chatReportParticipants[report.reportID] = report.participants. However, there is still a race condition here since the actions can fetch before the participants are updated.

Forcing the reportActions to wait until we are finished fetching the reportData is one way to fix this, but I'd rather not add that limitation unless we need to. Alternatively, we can make updates to the API to include participant data in one of the other API calls (GetHistory?). Bringing this to slack to chat about alternatives.

Jag96 committed:
update getIOUReport to use reportID

Jag96 commented:
I've created https://github.com/Expensify/Web-Expensify/pull/30506 and updated this PR so that `GetIOUReport` now takes `reportID` instead of `debtorEmail`. I've also updated the tests to account for the failing case as well as the Web branch required for testing.

In [slack](https://expensify.slack.com/archives/C011W8BJ9L6/p1616105476074000?thread_ts=1616104853.071700&cid=C011W8BJ9L6) we agreed that the ideal solution would be to have Auth or the API handle figuring out the participants for a chat report, and to return all the necessary data, which will be handled in https://github.com/Expensify/Expensify/issues/157722.

This is ready for review, but now requires https://github.com/Expensify/Web-Expensify/pull/30506 to work properly

tgolen commented on src/libs/actions/Report.js:
```diff
@@ -531,32 +531,22 @@ function getSimplifiedIOUReport(reportData) {
     // IOU's reportID, keep track of the IOU's reportID so we can use it to get the IOUReport data via `GetReportStuff`
     let iouReportID = 0;
     API.GetIOUReport({
-        debtorEmail: otherParticipants[0],
```
I think there needs to be a little bit more logic below in the `.then()` to handle the case where there is a reportID of 0 returned (which means we don't need to get reportStuff). I also think it should probably have a couple of `jsonCode === 200` checks as well since these report APIs can throw all kinds of errors that we aren't always prepared for.

Jag96 committed:
update onyx collection inside fetchChatReportsByIDs

Jag96 committed:
move def above usage

Jag96 commented:
Thanks for all the comments and discussion! Updated this based on the discussion in [slack](https://expensify.slack.com/archives/C011W8BJ9L6/p1616174958099100?thread_ts=1616104853.071700&cid=C011W8BJ9L6). Since we're already fetching reportActions as part of `fetchChatReportsByIDs` when we call `GetReportStuff`, we can grab the participants and reportActionList there, which is all that is required to call `GetReportIOU`.

I've also added some error handling for non-200 errors, empty reportList responses, and reportIDs of 0. I ran through the tests again and confirmed everything is working. This is ready for review again.

iwiznia commented on src/libs/actions/Report.js:
```diff
@@ -158,6 +158,97 @@ function getSimplifiedReportObject(report) {
+ * @param {Number} reportData.reportID
+ * @returns {Object}
+ */
+function getSimplifiedIOUReport(reportData) {
```
Why do we need to do this? Why not save the full reportData in onyx?

Jag96 replied:
From https://github.com/Expensify/Expensify/issues/157104, those are the only fields we wanted to store in the IOU collection. Do we want to store more of them? cc @Julesssss

Julesssss replied:
II don't see why we need too. If we do store the full report then we may as well merge the Onyx collection with the reports collection as previously discussed.

iwiznia replied:
True, but I am unsure why we did that, why transform the object into a slightly different one?

Julesssss replied:
Simple object == simple to understand was the idea, also to help reduce Onyx storage, which we believe will eventually be a limitation.

Having said that, I am not against reverting to the full object.

Jag96 replied:
I'd assume we'd only want to store the data we are going to use for the other IOU components. That's the reason we're using `getSimplifiedReportObject`:
```
* Only store the minimal amount of data in Onyx that needs to be stored
 * because space is limited
```

iwiznia commented on src/libs/actions/Report.js:
```diff
@@ -158,6 +158,97 @@ function getSimplifiedReportObject(report) {
+ * @param {String[]} participants
+ */
+function updateIOUReportData(report, participants) {
+    const reportActionList = lodashGet(report, ['reportActionList'], []);
```
```suggestion
    const reportActionList = report.reportActionList || [];
```

iwiznia commented on src/libs/actions/Report.js:
```diff
@@ -158,6 +158,97 @@ function getSimplifiedReportObject(report) {
+ */
+function updateIOUReportData(report, participants) {
+    const reportActionList = lodashGet(report, ['reportActionList'], []);
+    const containsIOUAction = _.any(reportActionList, reportAction => reportAction.action === 'IOU');
```
Let's make `IOU` a constant somewhere?

iwiznia commented on src/libs/actions/Report.js:
```diff
@@ -158,6 +158,97 @@ function getSimplifiedReportObject(report) {
+    }
+
+    // If we don't have one participant, this is not an IOU
+    if (participants.length !== 1) {
```
This should never happen really, should we log an alert?

iwiznia commented on src/libs/actions/Report.js:
```diff
@@ -158,6 +158,97 @@ function getSimplifiedReportObject(report) {
+ * @param {Object} report
+ * @param {String[]} participants
+ */
+function updateIOUReportData(report, participants) {
```
Why do we need to pass participants separated from report? Aren't the participants in the report data already?

Jag96 replied:
This was to avoid calling `getParticipantEmailsFromReport` twice, but the data set is small so I can call it twice to be more explicit.

iwiznia commented on src/libs/actions/Report.js:
```diff
@@ -158,6 +158,97 @@ function getSimplifiedReportObject(report) {
+    // IOU's reportID, keep track of the IOU's reportID so we can use it to get the IOUReport data via `GetReportStuff`
+    let iouReportID = 0;
+    API.GetIOUReport({
+        debtorEmail: participants[0],
```
How do you know that the first participant is the other user and not the current user?

Jag96 replied:
There will only be one participant here, since `getParticipantEmailsFromReport` calls `_without` and removes the currentUser.

iwiznia replied:
👍

iwiznia commented on src/libs/actions/Report.js:
```diff
@@ -158,6 +158,97 @@ function getSimplifiedReportObject(report) {
+        if (response.jsonCode !== 200) {
+            throw new Error(response.message);
+        } else if (iouReportID === 0) {
+            throw new Error('GetIOUReport returned a reportID of 0');
```
Why do we throw here? It's totally possible there's no IOU report that is unpaid

Jag96 replied:
I threw the error there to skip the `.then` chain below it. I can update this message to include `ignoring` or something similar to indicate it isn't an error.

iwiznia commented on src/libs/actions/Report.js:
```diff
@@ -158,6 +158,97 @@ function getSimplifiedReportObject(report) {
+    }).then((response) => {
+        if (response.jsonCode !== 200) {
+            throw new Error(response.message);
+        } else if (response.reports.length === 0) {
+            throw new Error('Empty reportList returned from Get_ReportStuff');
+        }
+
+        const iouReportData = response.reports[iouReportID];
+        if (!iouReportData) {
+            throw new Error(`No iouReportData found for reportID ${iouReportID}`);
+        }
```
I think you can unify these checks (or rather remove the length one)

Julesssss reviewed (dismissed):
Agree with a a few previously mentioned comments. 

Ran through the tests with no issues this time, Great work!

Jag96 committed:
get participants from report, clean up error handling

tgolen commented on src/CONST.js:
```diff
@@ -21,6 +21,9 @@ const CONST = {
     REPORT: {
         MAXIMUM_PARTICIPANTS: 8,
         REPORT_ACTIONS_LIMIT: 50,
+        REPORT_ACTION_TYPE: {
```
I think "REPORT" is redundant. I would suggest:

```
REPORT: {
    ACTIONS: {
        LIMIT: 50, 
        TYPE: {
            IOU: 'IOU',
        }
    },
}
```

tgolen commented on src/libs/actions/Report.js:
```diff
@@ -158,6 +158,101 @@ function getSimplifiedReportObject(report) {
+    const participants = getParticipantEmailsFromReport(report);
+    if (participants.length !== 1) {
+        Log.alert('[Report] Report with IOU action has more than 2 participants', true, {
+            reportID: report.reportID,
```
Maybe include the participants in this log? I think that would be helpful for troubleshooting it.

Jag96 commented:
@iwiznia @tgolen Updated

Jag96 committed:
update const, add participants to log

Jag96 commented:
Updated and re-tested

OSBotify commented:
:hand: This PR was not deployed to staging yet because QA is ongoing. It will be automatically deployed to staging after the next production release.

tgolen commented:
Sorry about not merging it! I was waiting for a test to finish... and apparently the GH page never updated when it was done.

Jag96 commented:
No worries! Those tests take a while!

isagoico commented:
@Jag96 Hello! This one looks like it's internal QA. If so, can you go ahead with the testing? When done please let me know so I check the PR of the list.

Jag96 commented:
Checked this off 👍

Julesssss commented:
This morning @trjExpensify and I noticed a potential regression with the Onyx changes:

The `iou` loading state values [that we initialize here](https://github.com/Expensify/Expensify.cash/blob/fe2355cc3b6b57d1ef19624e3756f30a712e549b/src/Expensify.js#L32) are [returned null here](https://github.com/Expensify/Expensify.cash/blob/fe2355cc3b6b57d1ef19624e3756f30a712e549b/src/pages/iou/IOUModal.js#L136) when opening the IOUModal. The data is valid within local storage and this is only reproducible about 1 in 4 attempts.

While the simple solution would be to add a conditional check to confirm the iou object is valid, we need to confirm whether this requires a fix, or is now expected behavior.

CC @kidroca 

https://user-images.githubusercontent.com/10736861/121342044-2e1dc380-c919-11eb-9670-fac9b03046e2.mov

Julesssss commented:
It looks like the cleanup PR supposedly [resolved this issue](https://github.com/Expensify/react-native-onyx/pull/79#issuecomment-856833531). But as far as I can tell, these commits are also on the staging build. So it seems to me that the issue is still occurring:

- [Tim's PR](https://github.com/Expensify/Expensify.cash/pull/3446) [bumps the Onyx commit](https://github.com/Expensify/Expensify.cash/pull/3446/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519L85-R85) to `4ff94b6c068582a75f50c6e159d3ff69fa123138`
- Commit `4ff94b6` [is more recent ](https://github.com/Expensify/react-native-onyx/commits/master)than the onyx cache fix
- Tim's PR was[ deployed to staging](https://github.com/Expensify/Expensify.cash/pull/3446#issuecomment-857283974)

Julesssss commented:
Moving this discussion [to the PR](https://github.com/Expensify/react-native-onyx/pull/76#issuecomment-857609265), so that @kidroca can participate
